panther_driver/src/panther_kineatics.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/python3
+
+import math
+from abc import abstractmethod
+
+import rospy
+
+from geometry_msgs.msg import Twist
+
+
+class PantherKinematics:
+    def __init__(self):
+        self.lin_x = 0
+        self.lin_y = 0
+        self.ang_z = 0
+        self.robot_width = 0
+        self.robot_length = 0
+        self.wheel_radius = 0  # Distance of the wheel center, to the roller center
+        self.encoder_resolution = 0
+        self.gear_ratio = 0
+        self.FR_enc_speed = 0  # front right encoder speed
+        self.FL_enc_speed = 0
+        self.RR_enc_speed = 0
+        self.RL_enc_speed = 0
+        self.robot_x_pos = 0
+        self.robot_y_pos = 0
+        self.robot_th_pos = 0
+        self.power_factor = 0
+        self.scale_factor_x = 0.25
+        self.scale_factor_y = 0.25
+        self.scale_factor_th = 0.125
+        self.cmd_vel_command_time = rospy.Time.now()
+
+    @abstractmethod
+    def forward_kinematics(self, x_data, y_data, th_data) -> None:
+        pass
+
+    @abstractmethod
+    def inverse_kinematics(self, wheel_FL_ang_vel, wheel_FR_ang_vel, wheel_RL_ang_vel, wheel_RR_ang_vel) -> float:
+        pass
+
+    def _get_motor_speed(self, wheel_front_right, wheel_front_left, wheel_rear_right, wheel_rear_left):
+        
+        _const_factor = self.power_factor * float(self.encoder_resolution * self.gear_ratio) / (2 * math.pi)
+
+        FR_enc_speed = _const_factor * float(wheel_front_right)
+        FL_enc_speed = _const_factor * float(wheel_front_left)
+        RR_enc_speed = _const_factor * float(wheel_rear_right)
+        RL_enc_speed = _const_factor * float(wheel_rear_left)
```

Comment:
```suggestion
        fl_enc_speed = _const_factor * float(wheel_front_left)
        fr_enc_speed = _const_factor * float(wheel_front_right)
        rl_enc_speed = _const_factor * float(wheel_rear_left)
        rr_enc_speed = _const_factor * float(wheel_rear_right)
```

According to REP 103 in **y** axis left is positive. I would suggest iterating over objects in manner from positive axis value to negative with respect to **y** axis first and **x** than. This results in order **fl**, **fr**, **rl**, **rr** . Also since those are not constants using capital letters in beginning of variables is not advised.